tests.py

```diff
@@ -7,14 +7,24 @@
 
 import os
 import unittest
+import threading
 from subprocess import Popen, PIPE
+from ipc.loader import load_plugin
 
 
 RULES_PATH = os.path.abspath(os.path.join('.', 'sample_rules'))
 
+def send(signal, value):
+    """Method overriden by the IPC plugin"""
+    pass
 
 class TestVSM(unittest.TestCase):
-    def run_vsm(self, name, input_data, expected_output, use_initial=True):
+
+    def setUp(self):
+        load_plugin('ipc.zeromq', use_receive=False)
+
+    def run_vsm(self, name, input_data, expected_output,
+                use_initial=True, use_ipc=None):
```

Comment:
I think the idea of using a flag to decide whether to use the IPC or not is a good one. However, duplicating all of the tests just to execute them with and without the flag is not. Have you considered adding this flag to the class, and at the bottom where you run unittest.main() you run it twice, once with it set to False, once with it set to True? Also, instead of a boolean the flag could contain the name of the module, so when we have the VSI one we can just add another set flag / run unittest.main() combo.